### ISSUE TITLE
AttributeError: 'Pokemon' object has no attribute 'num'

### ISSUE THREAD

**Issue by Zeper666** [PokemonGoF/PokemonGo-Bot#3785]
### Expected Behavior
work

### Actual Behavior
don't work more than 2 minute 

### Steps to Reproduce
start the bot and wait

### config.json 
```
{
    "auth_service": "ptc",
    "username": "coolusername",
    "password": "lolnope",
    "location": "48.842606,2.360102",
    "gmapkey": "acoolgmapkey",
    "encrypt_location": "",
    "tasks": [
      {
        "type": "HandleSoftBan"
      },
      {
        "type": "SleepSchedule",
        "config": {
          "enabled": false,
          "time": "22:54",
          "duration":"7:46",
          "time_random_offset": "00:24",
          "duration_random_offset": "00:43"
        }
      },
      {
        "type": "CollectLevelUpReward"
      },
      {
        "type": "IncubateEggs",
        "config": {
          "longer_eggs_first": true
        }
      },
      {
        "type": "UpdateLiveStats",
        "config": {
          "enabled": true,
          "min_interval": 10,
          "stats": ["uptime", "stardust_earned", "xp_earned", "xp_per_hour", "stops_visited"],
          "terminal_log": true,
          "terminal_title": true
        }
      },
      {
        "type": "TransferPokemon"
      },
      {
        "type": "NicknamePokemon",
        "config": {
          "enabled": false,
          "nickname_template": "{iv_pct}_{iv_ads}"
        }
      },
      {
        "type": "EvolvePokemon",
        "config": {
            "evolve_all": "Pidgey,Rattata,Weedle,Zubat",
            "first_evolve_by": "cp",
            "evolve_above_cp": 0,
            "evolve_above_iv": 0,
            "logic": "or",
            "evolve_speed": 1,
            "use_lucky_egg": false
        }
      },
      {
        "type": "RecycleItems",
        "config": {
          "min_empty_space": 15,
          "item_filter": {
            "Pokeball":       { "keep" : 20 },
			"Greatball":      { "keep" : 30 },
			"Ultraball":      { "keep" : 100 },
			"Masterball":     { "keep" : 999 },
            "Potion":         { "keep" : 0 },
            "Super Potion":   { "keep" : 0 },
            "Hyper Potion":   { "keep" : 0 },
            "Revive":         { "keep" : 0 },
            "Razz Berry":     { "keep" : 50 },
			"Max Revive":     { "keep" : 60 },
			"Max Potion":     { "keep" : 80 }
          }
        }
      },
      {
        "type": "CatchVisiblePokemon"
      },
      {
        "type": "CatchLuredPokemon"
      },
      {
        "type": "SpinFort"
      },
      {
        "type": "MoveToFort",
        "config": {
            "lure_attraction": true,
            "lure_max_distance": 2000
        }
      },
      {
        "type": "FollowSpiral",
        "config": {
          "diameter": 4,
          "step_size": 70
        }
      }
    ],
    "map_object_cache_time": 5,
    "forts": {
      "avoid_circles": true,
      "max_circle_size": 50,
      "cache_recent_forts": true
    },
    "websocket_server": false,
    "walk": 14.16,
    "action_wait_min": 0,
    "action_wait_max": 0,
    "debug": false,
    "test": false,
    "health_record": true,
    "location_cache": true,
    "distance_unit": "km",
    "reconnecting_timeout": 15,
    "catch_randomize_reticle_factor": 1.0,
    "catch_randomize_spin_factor": 1.0,
    "min_ultraball_to_keep": 10,
    "logging_color": true,
    "catch": {
      "any": {"catch_above_cp": 0, "catch_above_iv": 0, "logic": "or"},
      "// Example of always catching Rattata:": {},
      "// Rattata": { "always_catch" : true }
    },
    "catch_throw_parameters": {
      "excellent_rate": 0.1,
      "great_rate": 0.5,
      "nice_rate": 0.3,
      "normal_rate": 0.1,
      "spin_success_rate" : 0.6
    },
    "release": {
      "any": {"release_below_cp": 1500, "release_below_iv": 0.7, "logic": "or"},
      "// Example of always releasing Rattata:": {},
      "// Rattata": {"always_release": true},
      "// Example of keeping 3 stronger (based on CP) Pidgey:": {},
      "// Pidgey": {"keep_best_cp": 3},
      "// Example of keeping 2 stronger (based on IV) Zubat:": {},
      "// Zubat": {"keep_best_iv": 2},
      "// Also, it is working with any": {},
      "// any": {"keep_best_iv": 3},
      "// Example of keeping the 2 strongest (based on CP) and 3 best (based on IV) Zubat:": {},
      "any": {"keep_best_cp": 2, "keep_best_iv": 2}
    },
    "vips" : {
         "Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "any": {"catch_above_cp": 1200, "catch_above_iv": 0.9, "logic": "or" },
        "Lapras": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},

        "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
        "Mewtwo": {},
        "Dragonite": {},
        "Snorlax": {},
        "// Mew evolves to Mewtwo": {},
        "Mew": {},
        "Arcanine": {},
        "Vaporeon": {},
        "Gyarados": {},
        "Exeggutor": {},
        "Muk": {},
        "Weezing": {},
        "Flareon": {}

    }
}

```
### Error message
 
```
2016-08-12 21:35:57,878 [       cli] [INFO]
2016-08-12 21:35:57,878 [       cli] [INFO] Ran for 0:02:10
2016-08-12 21:35:57,880 [       cli] [INFO] Total XP Earned: 1415  Average: 39159.61/h
2016-08-12 21:35:57,881 [       cli] [INFO] Travelled 0.05km
2016-08-12 21:35:57,881 [       cli] [INFO] Visited 6 stops
2016-08-12 21:35:57,881 [       cli] [INFO] Encountered 5 pokemon, 4 caught, 0 released, 1 evolved, 0 never seen before
2016-08-12 21:35:57,881 [       cli] [INFO] Threw 11 pokeballs
2016-08-12 21:35:57,881 [       cli] [INFO] Earned 400 Stardust
2016-08-12 21:35:57,881 [       cli] [INFO]
2016-08-12 21:35:57,881 [       cli] [INFO] Highest CP Pokemon: Psyduck [CP: 579] [IV: 3/12/6] Potential: 0.47
2016-08-12 21:35:57,882 [       cli] [INFO] Most Perfect Pokemon: Rattata [CP: 70] [IV: 12/9/9] Potential: 0.67
Traceback (most recent call last):
  File "pokecli.py", line 602, in <module>
    main()
  File "pokecli.py", line 103, in main
    bot.tick()
  File "C:\PokemonGo-Bot-dev\pokemongo_bot\__init__.py", line 482, in tick
    if worker.work() == WorkerResult.RUNNING:
  File "C:\PokemonGo-Bot-dev\pokemongo_bot\cell_workers\catch_visible_pokemon.py", line 49, in work
    self.catch_pokemon(self.bot.cell['catchable_pokemons'].pop(0))
  File "C:\PokemonGo-Bot-dev\pokemongo_bot\cell_workers\catch_visible_pokemon.py", line 70, in catch_pokemon
    return_value = worker.work()
  File "C:\PokemonGo-Bot-dev\pokemongo_bot\cell_workers\pokemon_catch_worker.py", line 102, in work
    self._do_catch(pokemon, encounter_id, catch_rate_by_ball, is_vip=is_vip)
  File "C:\PokemonGo-Bot-dev\pokemongo_bot\cell_workers\pokemon_catch_worker.py", line 370, in _do_catch
    'pokemon_id': pokemon.num
AttributeError: 'Pokemon' object has no attribute 'num'
2016-08-12 21:35:57,904 [sentry.errors] [ERROR] Sentry responded with an error: 'ascii' codec can't decode byte 0x9c in position 1: ordinal not in range(128) (url: https://app.getsentry.com/api/90254/store/)
Traceback (most recent call last):
  File "C:\Python27\lib\site-packages\raven\transport\threaded.py", line 174, in send_sync
    super(ThreadedHTTPTransport, self).send(data, headers)
  File "C:\Python27\lib\site-packages\raven\transport\http.py", line 47, in send
    ca_certs=self.ca_certs,
  File "C:\Python27\lib\site-packages\raven\utils\http.py", line 66, in urlopen
    return opener.open(url, data, timeout)
  File "C:\Python27\lib\site-packages\future\backports\urllib\request.py", line 494, in open
    response = self._open(req, data)
  File "C:\Python27\lib\site-packages\future\backports\urllib\request.py", line 512, in _open
    '_open', req)
  File "C:\Python27\lib\site-packages\future\backports\urllib\request.py", line 466, in _call_chain
    result = func(*args)
  File "C:\Python27\lib\site-packages\raven\utils\http.py", line 46, in https_open
    return self.do_open(ValidHTTPSConnection, req)
  File "C:\Python27\lib\site-packages\future\backports\urllib\request.py", line 1284, in do_open
    h.request(req.get_method(), req.selector, req.data, headers)
  File "C:\Python27\lib\httplib.py", line 1057, in request
    self._send_request(method, url, body, headers)
  File "C:\Python27\lib\httplib.py", line 1097, in _send_request
    self.endheaders(body)
  File "C:\Python27\lib\httplib.py", line 1053, in endheaders
    self._send_output(message_body)
  File "C:\Python27\lib\httplib.py", line 895, in _send_output
    msg += message_body
UnicodeDecodeError: 'ascii' codec can't decode byte 0x9c in position 1: ordinal not in range(128)
2016-08-12 21:35:57,907 [sentry.errors.uncaught] [ERROR] [u"AttributeError: 'Pokemon' object has no attribute 'num'", u'  File "pokecli.py", line 602, in <module>', u'  File "pokecli.py", line 103, in main', u'  File "C:\\PokemonGo-Bot-dev\\pokemongo_bot\\__init__.py", line 482, in tick', u'  File "C:\\PokemonGo-Bot-dev\\pokemongo_bot\\cell_workers\\catch_visible_pokemon.py", line 49, in work', u'  File "C:\\PokemonGo-Bot-dev\\pokemongo_bot\\cell_workers\\catch_visible_pokemon.py", line 70, in catch_pokemon', u'  File "C:\\PokemonGo-Bot-dev\\pokemongo_bot\\cell_workers\\pokemon_catch_worker.py", line 102, in work', u'  File "C:\\PokemonGo-Bot-dev\\pokemongo_bot\\cell_workers\\pokemon_catch_worker.py", line 370, in _do_catch']
```

### Other Information
OS: win10 x64
Branch: dev

**Comment by pmquan** [comment]
Fixed in #3780

**Comment by GitPupper** [comment]
Thanks for helping out @pmquan
